lib/src/optional.dart

```diff
@@ -84,14 +72,13 @@ class Optional<T> {
   String toString() => '$Optional($_value)';
 
   @override
-  bool operator ==(Object other) =>
-      identical(this, other) || other is Optional && _value == other._value;
+  bool operator ==(Object other) => identical(this, other) || other is Optional && _value == other._value;
 
   @override
   int get hashCode => _value?.hashCode ?? 0;
 
   /// Avoid using this property.
-  T get unsafe => _value;
+  T? get unsafe => _value;
 
-  final T _value;
+  final T? _value;
```

Comment:
I made the minimal changes to make it null-safe. For a next version you could consider dropping `unsafe` and making `value` public.